internal/verifier/compare.go

```diff
@@ -308,16 +315,26 @@ func (verifier *Verifier) getFetcherChannels(
 		}
 
 		return err
-	})
+	}
 
-	return srcChannel, dstChannel
+	return srcChannel, dstChannel, readSrcCallback, readDstCallback
 }
 
-func iterateCursorToChannel(ctx context.Context, cursor *mongo.Cursor, writer chan<- bson.Raw) error {
+func iterateCursorToChannel(
+	ctx context.Context,
+	state *retry.FuncInfo,
+	cursor *mongo.Cursor,
+	writer chan<- bson.Raw,
+) error {
 	for cursor.Next(ctx) {
+		state.NoteSuccess()
 		writer <- slices.Clone(cursor.Current)
 	}
 
+	if cursor.Err() == nil {
+		state.NoteSuccess()
```

Comment:
Looks like the retryer refreshes `lastResetTime` once the error group finishes. Is it redundant to call `NoteSuccess` before the callback function returns?

Reply:
Probably. Removed.